src/main/java/io/mosip/iiitb/odte/utils/HMACUtil2.java

```diff
@@ -0,0 +1,63 @@
+package io.mosip.iiitb.odte.utils;
+
+import java.nio.ByteOrder;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * This class defines the alternate safer HMAC Util to be used in MOSIP Project.
+ * The HMAC Util is implemented using desired methods of MessageDigest class of
+ * java security package
+ *
+ * @author Sasikumar Ganesan
+ *
+ * @since 1.1.4
+ */
+public final class HMACUtil2 {
```

Comment:
Instead of copy pasting the code, can we not use as library instead ?

Reply:
Even I wanted to use the library and that was the approach we initially took.There were dependency errors popping up, and we were short on time, so we decided to copy paste the code, so that we could test and verify that the service was working as expected and decided we'll come back to this once the code is known is work.